### PR TITLE
Angular 15 and Webpack 5 upgrade

### PR DESCRIPTION
**Pull request by nvladimirovi**
Implement new Custom Angular builder, for Angular 15, used to build VMware Cloud Director UI Plugins.

The change includes:
- Custom Angular Builder
- Custom PostCSS Plugin
- Custom Webpack Plugn (for concatenation of files)
- Merge @vcd/angular-client with @vcd/sdk project
- Deletion of @vcd/sdk/i18n (Implemented by @iilieva)
- Deletion of Angular Schematics (they are not used, we have CLI for that)
- Update READMEs

### CODE REVIEW

**Review by stoychevl — CHANGES_REQUESTED** [review]
We need to come up with a solution for handling API Client Negotiated versions, because currently the hardcoded versions in the client lack behind the new VCD API versions. This will cause some plugins in the future to fail. We have two approaches:

- Maintain the Supported versions list 
- Always chose highest supported API version from api/versions response

The current implementation of "actAs()" method, doesn't return a clone of the VCD Api client, but rather a mutated reference of the client. This makes every consecutive call of the original client to use the latest set context. We can:

- actAs() returns a copy of the vcdApi client with the changed context
- pass actAs as a optional property of the CRUD http functions in the service

The current implementation of setVersion can only be used after the initialisation of the apiClient, thus the client always uses the negotiated maximum version for the initial API session calls.

- Pass apiClient version from service init()
- Pass apiClient version via .forRoot() in the SKD module.

File: packages/angular/projects/vcd/plugin-builders/src/lib/base/index.ts [hunk 1]
```diff
@@ -47,239 +30,242 @@ export const defaultExternals = {
...
+
+        // Load manifest file
+        const manifestJsonPath = path.join(`${config.context}/src/public`, 'manifest.json');
+        const manifest: ExtensionManifest = JSON.parse(fs.readFileSync(manifestJsonPath, 'utf-8'));
+        const manifestOriginalContent: string = fs.readFileSync(manifestJsonPath, 'utf-8');
+        
+        // Patch the main.ts file to point to the plugin which will be compiled
+        let [modulePath, moduleName] = options.modulePath.split('#');
+
+        if (options.enableRuntimeDependecyManagement) {
```
Comment (stoychevl) [inline packages/angular/projects/vcd/plugin-builders/src/lib/base/index.ts]: Please fix the type enableRuntimeDependecyManagement -> enableRuntimeDependencyManagement

**Comment by stoychevl** [comment]
We need to come up with a solution for handling API Client Negotiated versions, because currently the hardcoded versions in the client lack behind the new VCD API versions. This will cause some plugins in the future to fail.

**Review by stoychevl — APPROVED** [review]
It all makes sense, however we also need to delete all legacy vcd-ext CLI related code and the CARE package implementation.

File: packages/plugin-builders/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+- Angular Builder
+- Webpack
+- PostCSS Plugins
+
+We are goin to cover each piece.
+### Angular Builder
+#### Description
+We implement our custom Angular Builder to build VMware Cloud Director UI Plugins, hence the name - Plugin Builder.
+
+We implement it in the standard Angular documented way,
```
Comment (iilieva) [inline packages/plugin-builders/README.md:36]: This information is redundant.

File: packages/plugin-builders/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+#### Useful References:
+- [Angular CLI builders](https://angular.io/guide/cli-builder)
+- [Angular Builders GitHub](https://github.com/just-jeb/angular-builders)
+
+
+### Webpack
+#### Description
+We use webpack's feature `splitChunks` and `externals` to isolate the npm dependencies from your UI Plugin code, and therefore we inject in the final bundle only the source code of the UI Plugin plus custom libraries.
+
+All npm dependencis in the `externals` list will be provided to your UI Plugin by VMware Cloud Director at runtime, this boosts the loading time of your UI Plugin a lot.
```
Comment (iilieva) [inline packages/plugin-builders/README.md:48]: nit: a lot -> significantly

File: packages/plugin-builders/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+We leverage Webpack's `DefinePlugin`, to replace keywords in the final JS bundle. Most often we use to perform JS isolation in special type of UI Plguins called Hyper. Regular UI Plugins doesn't leverage the `DefinePlugin`.
```
Comment (iilieva) [inline packages/plugin-builders/README.md:50]: nit: doesn't -> do not

File: packages/plugin-builders/README.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+- [Webpack SplitChunksPlugin](https://webpack.js.org/plugins/split-chunks-plugin/)
+- [Externals](https://webpack.js.org/configuration/externals/)
+
+### PostCSS Plugins
+#### Description
+We have 2 types of UI Plugins:
+- Regular, implemented by 3rd parties
+- Hyper, this type of UI Plugins support the regular with: JS isolation, CSS isolation and more.
+
+Now, the PostCSS Plugins are not used by the Regular UI Plugins in build time, it's used only by the Hyper ones.
```
Comment (iilieva) [inline packages/plugin-builders/README.md:63]: nit: The PostCss plugins are used at build time only by the Hyper plugins. The regular UI plugins do not utilize them.

File: packages/plugin-builders/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "@vcd/plugin-builders",
+  "version": "15.0.0",
+  "description": "Angular CLI-compatible tools for building Cloud Director plugins.",
+  "author": "VMware",
+  "license": "BSD-2-Clause",
+  "builders": "builders.json",
+  "dependencies": {
+    "@angular-devkit/architect": ">=0.1500.0 < 0.1600.0",
+    "@angular-devkit/build-angular": "^15.0.0",
```
Comment (iilieva) [inline packages/plugin-builders/package.json:10]: The same suggestions for fixed versions here.

**Review by iilieva — APPROVED** [review]

**Review by irahov — APPROVED** [review]

File: packages/angular/package.json [hunk 7]
```diff
@@ -1,74 +1,39 @@
 {
-  "name": "@vcd/angular-sdk",
-  "version": "0.13.0-alpha.4",
+  "name": "angular",
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (nvladimirovi) [reply packages/angular/package.json:3]: That is not the package version, this is a version that isn't used by NPM or anything other, that's just the "holder" project.

File: packages/angular/projects/vcd/sdk/src/client/client/vcd.api.client.spec.data.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+                id: 'urn:vcloud:site:4f569ca4-2a1f-457c-b587-21339358a38a'
+            },
+            org: {
+                name: 'System',
+                id: 'urn:vcloud:org:a93c9db9-7471-3192-8d09-a8f7eeda85f9'
+            },
+            restApiEndpoint: 'https://cassini.eng.vmware.com',
+            uiEndpoint: 'https://cassini.eng.vmware.com',
+            apiVersion: '36.0'
+        },
```
[the remark this answers is not in the document]
Comment (nvladimirovi) [reply packages/angular/projects/vcd/sdk/src/client/client/vcd.api.client.spec.data.ts:32]: This is a test that we don't run currently, we'll need to implement new ones, or refactor the old.

**Pull request merged**